drivers/timer/Kconfig

```diff
@@ -234,6 +234,15 @@ config RISCV_MACHINE_TIMER
 	  This module implements a kernel device driver for the generic RISCV machine
 	  timer driver. It provides the standard "system clock driver" interfaces.
 
+config RISCV_MACHINE_TIMER_SYSTEM_CLOCK_DIVIDER
+	int "RISCV Machine Timer system clock divider"
+	default 0
```

Comment:
non-blocking suggestion: a divider can never be `0`, so defaulting to `1` would make more sense (and then just subtract 1 on the shift operator).

Reply:
I will apply @nandojve 's suggestion that defines the divider value in devicetree.
https://github.com/zephyrproject-rtos/zephyr/pull/34970#discussion_r762211524

Changes to use the value as a simple divide factor (not 2^n).
In this case, divided by zero can occur.
So it changes the default also to 1.
And add static assert to detect the value, not 0 or not a power of 2.